cranelift/filetests/filetests/isa/x64/simd-arith-avx.clif

```diff
@@ -1648,8 +1646,6 @@ block0(v0: i8x16, v1: i32):
 ;   addb %al, (%rax)
 ;   addb %al, (%rax)
 ;   addb %al, (%rax)
-;   addb %al, (%rax)
-;   addb %bh, %bh
```

Comment:
Looking at this CLIF diffs, I see some potential problems: removed instructions, different immediates, potential `and` -> `or` miscompilations... This bears another look.

Reply:
(Pretty sure those `add` -> `or` "miscompilations" were just bits changing in the constant pool... not sure why they're changing, though).

Reply:
This one looks like 3 bytes were shaved off the encoding (the 0x19 => 0x16 in the diff), are the new assembler methods perhaps slightly more optimal in encoding size?

Reply:
Yeah, that's what I concluded.